relay-pii/src/processor.rs

```diff
@@ -1177,11 +1290,65 @@ mod tests {
                     .unwrap()
                     .formatted
                     .value()
-                    .is_none()
+                    .is_some()
             );
         }
     }
 
+    #[test]
+    fn test_logentry_formatted_never_fully_filtered() {
+        // Test that logentry.formatted is never completely filtered even with aggressive PII rules
+        let config = serde_json::from_str::<PiiConfig>(
+            r#"
+        {
+            "applications": {
+                "$logentry.formatted": ["@anything:remove"]
+            }
+        }
+        "#,
+        )
+        .unwrap();
+
+        let mut event = Annotated::new(Event {
+            logentry: Annotated::new(LogEntry {
+                formatted: Annotated::new(
+                    "User john.doe@company.com failed login with card 4111-1111-1111-1111"
+                        .to_owned()
+                        .into(),
+                ),
+                ..Default::default()
+            }),
+            ..Default::default()
+        });
+
+        let mut processor = PiiProcessor::new(config.compiled());
+        process_value(&mut event, &mut processor, ProcessingState::root()).unwrap();
+
+        let formatted_value = event
+            .value()
+            .unwrap()
+            .logentry
+            .value()
+            .unwrap()
+            .formatted
+            .value();
+
+        // Should never be None (completely removed)
+        assert!(formatted_value.is_some());
+
+        // Should have smart scrubbing applied (emails and credit cards scrubbed)
+        let formatted_str = formatted_value.unwrap().as_ref();
+        assert!(formatted_str.contains("[Email]"));
+        assert!(formatted_str.contains("[CreditCard]"));
+
+        // Should not be "[Filtered]"
+        assert_ne!(formatted_str, "[Filtered]");
+
+        // Should preserve context
+        assert!(formatted_str.contains("User"));
+        assert!(formatted_str.contains("failed login"));
```

Comment:
I think it might be easier to tell what the end result should look like with a snapshot.